gap/PolygonalComplexes/animating.gd

```diff
@@ -0,0 +1,537 @@
+#############################################################################
+##
+##  SimplicialSurface package
+##
+##  Copyright 2012-2019
+##    Markus Baumeister, RWTH Aachen University
+##    Alice Niemeyer, RWTH Aachen University
+##    Jens Brandt, RWTH Aachen University
+##
+## Licensed under the GPL 3 or later.
+##
+#############################################################################
+
+#! @Chapter Animating surfaces with Java Script
+#! @ChapterLabel AnimationJavaScript
+#!
+#! This chapter deals with animating simplicial surfaces via Java Script
+#! using three.js, see <URL>https://threejs.org/</URL>.
+
+#! @Section Introduction and Quick Start
+#! @SectionLabel LabelIntroductionAndQuickStartAnimating
+
+#! This section contains a minimal example by animating an octahedron and after that shows the general strategy to use the functions of this chapter.
+#! To construct an octahedron we only need to specify the 3D-coordinates of the vertices.
+#! @BeginExampleSession
+#! gap> oct := Octahedron();;
+#! gap> verticesPositions := [
+#! > [ 0, 0, Sqrt(2.) ],
+#! > [ 1, 1, 0 ],
+#! > [ 1, -1, 0 ],
+#! > [ -1, -1, 0 ],
+#! > [ -1, 1, 0 ],
+#! > [ 0, 0, -Sqrt(2.) ] ];;
+#! gap> printRecord := SetVertexCoordiantes3D(oct, verticesPositions, rec());;
+# rec( vertexCoordinates3D := [ [ 0, 0, 1.41421 ], [ 1, 1, 0 ], [ 1, -1, 0 ], [ -1, -1, 0 ], [ -1, 1, 0 ], [ 0, 0, -1.41421 ] ] )
+#! gap> DrawSurfaceToJavaScript(oct, "octahedron.html", printRecord);;
+#! @EndExampleSession
+#! Now your working directory should contain a file "octahedron.html". Below is shown an image of the animation.
+#! @InsertChunk Example_OctahedronAnimating
+
+#! The philosophy of the code presented in this chapter is the following.
+#! The main function <K>DrawSurfaceToJavaScript</K> (Section <Ref Sect="Section_LabelCoordinatesAndCoreFunctionality"/>) animates the surfaces using the configuration saved in the so called  print record, e.g. the 3D-coordinates (3D-coordinates are defined as a list with 3 entries of floats) of the vertices. It is necessary that the print record contains the 3D-coordinates of each vertex of the surface which should be animated. Based on the 3D-coordinates of the vertices the surface is placed in the animation.
```

Comment:
"print record" should be emphasized the first time.
.. the 3D-coordinates (lists with three float entries) ..
.. Based on the 3D-coordinates of the vertices, ...
Therefore,